src/sentry/api/endpoints/organization_integration_repository_project_path_configs.py

```diff
@@ -90,31 +104,96 @@ def update(self, instance, validated_data):
         return self.instance
 
 
+class NullableOrganizationIntegrationMixin:
+    def get_organization_integration(self, organization, integration_id):
+        try:
+            return OrganizationIntegration.objects.get(
+                integration_id=integration_id,
+                organization=organization,
+            )
+        except OrganizationIntegration.DoesNotExist:
+            raise Http404
+
+    def get_project(self, organization, project_id):
+        try:
+            return Project.objects.get(organization=organization, id=project_id)
+
+        except Project.DoesNotExist:
+            raise Http404
+
+
 class OrganizationIntegrationRepositoryProjectPathConfigEndpoint(
-    OrganizationIntegrationBaseEndpoint
+    OrganizationEndpoint, NullableOrganizationIntegrationMixin
 ):
     permission_classes = (OrganizationIntegrationsPermission,)
 
-    def get(self, request, organization, integration_id):
+    def get(self, request, organization):
         """
-        Get the list of repository project path configs in an integration
+        Get the list of repository project path configs
+
+        :pparam string organization_slug: the slug of the organization the
+                                          team should be created for.
+        :queryparam int integrationId: the optional integration id.
+        :queryparam int projectId: the optional project id.
+        :auth: required
         """
-        org_integration = self.get_organization_integration(organization, integration_id)
 
-        # front end handles ordering
-        repository_project_path_configs = RepositoryProjectPathConfig.objects.filter(
-            organization_integration=org_integration
-        )
+        integration_id = request.GET.get("integrationId")
+        project_id = request.GET.get("projectId")
 
+        if not integration_id and not project_id:
+            return self.respond(
+                {"detail": 'Missing valid "projectId" or "integrationId"'},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        queryset = RepositoryProjectPathConfig.objects.all()
+
+        if integration_id:
+            # Return early with 404
```

Comment:
@NisanthanNanthakumar I feel like this comment needs more detail (that get_organization_integration will raise a 404 if no org_integraiton is found) since we aren't actually returning early in this function.